modules/docs/markdown/02.1-serialisation/01-serialisation.md

```diff
@@ -0,0 +1,118 @@
+---
+sidebar_label: Serialisation overview
+title: Serialisation overview
+---
+
+The code generated by Smithy4s is strictly **protocol agnostic**. One implication is that the data-types generated by Smithy4s are not tied to any particular serialisation format or third-party library. Instead, Smithy4s generates an instance of a `smithy4s.schema.Schema` for each data-type (see [the relevant section](../05-design/02-schemas.md)). From this schema can be derived encoders and decoders for virtually any serialisation format.
+
+Smithy4s provides opt-in modules implementing serialisation in a bunch of formats, including `JSON`, `XML` and `Protobuf`. The modules cross-compile to all combinations of platforms (JVM/JS/Native) and scala-versions supported by Smithy4s.
+
+### Document (JSON-like adt)
+
+The `smithy4s-core` module provides a `smithy4s.Document` datatype that is used in code-generation when [document](https://smithy.io/2.0/spec/simple-types.html#document) shapes are used in smithy. `Document` is effectively a JSON ADT, and can be easily converted to from other ADTs, such as the one provided by the [Circe library](https://circe.github.io/circe/).
+
+`Document` also comes with its own Encoder and Decoder construct, for which instances can be derived for every datatype generated by Smithy4s.
+
+```scala mdoc:reset
+import smithy4s.example.hello.Person
+import smithy4s.Document
+
+val personEncoder = Document.Encoder.fromSchema(Person.schema)
+val personDocument = personEncoder.encode(Person(name = "John Doe"))
+
+val personDecoder = Document.Decoder.fromSchema(Person.schema)
+val maybePerson = personDecoder.decode(personDocument)
+```
+
+By default, smithy4s Documents abide by the same semantics as `smithy4s-json` (see section below).
+
+It is worth noting that, although `Document` is isomorphic to a JSON ADT, its `.toString` is not valid JSON. Likewise, the `smithy4s-core` module does not contain logic to parse JSON strings into Documents. In order to read/write Documents from/to JSON strings, you need the `smithy4s-json` module. The `smithy4s.json.Json` entry-point contains methods that work with Documents.
+
+### JSON
+
+The `smithy4s-json` module provides [jsoniter-based](https://github.com/plokhotnyuk/jsoniter-scala) encoders/decoders that can read/write generated data-types from/to JSON bytes/strings, without an intermediate JSON ADT. The performance of this module is very competitive are [very competitive](https://plokhotnyuk.github.io/jsoniter-scala/) in the Scala ecosystem.
+
+This module is provided at the following coordinates :
+
+```
+sbt : "com.disneystreaming.smithy4s"  % "smithy4s-json" % "@VERSION@"
```

Comment:
**issue:** `%%` instead of `%`, also there are extra spaces.

suggestion: UI-wise I think we could improve upon these code snippets by splitting them in two, in addition we could use more advanced Docusaurus features such as [mdx tabs](https://docusaurus.io/docs/markdown-features/tabs). They're nice in the fact that you can make them synchronized, so that picking Mill once updates all the visible tab groups. Example: https://scala-php.org/docs/getting-started/ (sbt/mill tab groups are synced)

Reply:
https://github.com/disneystreaming/smithy4s/issues/1471

Reply:
SGTM